From cfc38dc4ccccb5fd218eeb560bde21a42f381b91 Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Sat, 7 May 2022 15:00:06 +0800
Subject: [PATCH] sw64: clean up unused pci iounmap operation
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: cfc38dc4ccccb5fd218eeb560bde21a42f381b91
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

For sw64, IO address is mapped by direct mapping without pages.
It doesn't have to do anything to unmap PCI MMIO. The result is
that __is_mmio helper can be removed.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/io.h | 8 --------
 arch/sw_64/kernel/pci.c     | 4 ----
 2 files changed, 12 deletions(-)

diff --git a/arch/sw_64/include/asm/io.h b/arch/sw_64/include/asm/io.h
index 6796c64f94ae..cb6c7c1f4a6b 100644
--- a/arch/sw_64/include/asm/io.h
+++ b/arch/sw_64/include/asm/io.h
@@ -218,14 +218,6 @@ static inline int __is_ioaddr(unsigned long addr)
 
 #define __is_ioaddr(a)  __is_ioaddr((unsigned long)(a))
 
-static inline int __is_mmio(const volatile void __iomem *xaddr)
-{
-	unsigned long addr = (unsigned long)xaddr;
-
-	return (addr & 0x100000000UL) == 0;
-}
-
-
 
 #define ioread16be(p) be16_to_cpu(ioread16(p))
 #define ioread32be(p) be32_to_cpu(ioread32(p))
diff --git a/arch/sw_64/kernel/pci.c b/arch/sw_64/kernel/pci.c
index 44264e3da18f..77bbd1b68176 100644
--- a/arch/sw_64/kernel/pci.c
+++ b/arch/sw_64/kernel/pci.c
@@ -358,12 +358,8 @@ asmlinkage long sys_pciconfig_iobase(long which, unsigned long bus, unsigned lon
 	return -EOPNOTSUPP;
 }
 
-/* Destroy an __iomem token.  Not copied from lib/iomap.c.  */
-
 void pci_iounmap(struct pci_dev *dev, void __iomem *addr)
 {
-	if (__is_mmio(addr))
-		iounmap(addr);
 }
 EXPORT_SYMBOL(pci_iounmap);
 
-- 
2.33.0

